src/index.js

```diff
@@ -24,6 +24,11 @@ const isNestedArray = arr => bool(
 
 const isMap = m => bool(Object.prototype.toString.call(m) === '[object Map]')
 
+// check if consumer is yielding our effect to immediatly cause the generator function to throw an error
+const shouldThrowError = obj => bool(obj && Object.keys(obj).includes('@THROW'))
```

Comment:
I think you could just do an `typeof obj === 'object'` and `obj.hasOwnProperty('magic-key')`